src/fidesops/ops/api/v1/endpoints/consent_request_endpoints.py

```diff
@@ -0,0 +1,246 @@
+            & (
+                ProvidedIdentity.privacy_request  # pylint: disable=singleton-comparison
+                == None
+            )
+        ),
+    ).first()
+
+    if not identity:
+        provided_identity_data = {
+            "privacy_request_id": None,
+            "field_name": "email",
+            "hashed_value": ProvidedIdentity.hash_value(data.email),
+            "encrypted_value": {"value": data.email},
+        }
+        identity = ProvidedIdentity.create(db, data=provided_identity_data)
+
+    consent_request_data = {
+        "provided_identity_id": identity.id,
+    }
+    consent_request = ConsentRequest.create(db, data=consent_request_data)
+    try:
+        verification_code = send_verification_code_to_user(
+            db, consent_request, data.email
+        )
+    except EmailDispatchException as exc:
+        logger.error("Error sending the verification code email: %s", Pii(str(exc)))
+        raise HTTPException(
+            status_code=HTTP_500_INTERNAL_SERVER_ERROR,
+            detail="Error sending the verification code email",
+        )
+    return ConsentRequestResponse(
+        identity=data,
+        consent_request_id=consent_request.id,
+        verification_code=verification_code,
+    )
+
+
+@router.post(
+    CONSENT_REQUEST_VERIFY,
+    status_code=HTTP_200_OK,
+    response_model=ConsentPreferences,
+)
+def consent_request_verify(
+    *,
+    consent_request_id: str,
+    db: Session = Depends(get_db),
+    data: VerificationCode,
+) -> ConsentPreferences:
+    """Verifies the verification code and returns the current consent preferences if successful."""
+    provided_identity = _get_consent_request_and_provided_identity(
+        db=db, consent_request_id=consent_request_id, verification_code=data.code
+    )
+
+    if not provided_identity.hashed_value:
+        raise HTTPException(
+            status_code=HTTP_404_NOT_FOUND, detail="Provided identity missing email"
+        )
+
+    consent = Consent.filter(
+        db=db, conditions=Consent.provided_identity_id == provided_identity.id
+    ).all()
+
+    return _prepare_consent_preferences(consent)
+
+
+@router.patch(
+    CONSENT_REQUEST_PREFERENCES,
+    status_code=HTTP_200_OK,
+    response_model=ConsentPreferences,
+)
+def set_consent_preferences(
+    *,
+    consent_request_id: str,
+    db: Session = Depends(get_db),
+    data: ConsentPreferencesWithVerificationCode,
+) -> ConsentPreferences:
+    """Verifies the verification code and saves the user's consent preferences if successful."""
+    provided_identity = _get_consent_request_and_provided_identity(
+        db=db,
+        consent_request_id=consent_request_id,
+        verification_code=data.code,
+    )
+
+    if not provided_identity.hashed_value:
+        raise HTTPException(
+            status_code=HTTP_404_NOT_FOUND, detail="Provided identity missing email"
+        )
+
+    for preference in data.consent:
```

Comment:
What is the expected behavior if the user has two Consents saved for two different data uses, and then they want to delete one of their Consents? It's not currently possible, they can update existing Consent records here or add new ones, but not delete.  If they pass in a list of one Consent here, the existing one is not removed.  

Should it be?

Reply:
From my understanding the consent values will come from a set list. So they can't delete them, but they can change from `True` to `False`.